driver/tab_completion.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import importlib.util
+from pathlib import Path
+import subprocess
+import sys
+
+from . import util
+
+try:
+    import argcomplete
+    HAS_ARGCOMPLETE = True
+except ImportError:
+    HAS_ARGCOMPLETE = False
+
+
+def complete_build_arg(prefix, parsed_args, **kwargs):
+    if parsed_args.debug:
+        argcomplete.warn("The option --debug is an alias for --build=debug. Do no specify both --debug and --build.")
+        exit(1)
+
+    builds_folder = Path(util.REPO_ROOT_DIR) / "builds"
+    if not builds_folder.exists():
+        argcomplete.warn("No build exists.")
+        exit(1)
+    return [p.name for p in builds_folder.iterdir() if p.is_dir()]
+
+
+def complete_planner_args(prefix, parsed_args, **kwargs):
+    if parsed_args.build and parsed_args.debug:
+        argcomplete.warn("The option --debug is an alias for --build=debug. Do no specify both --debug and --build.")
+        exit(1)
+
+    build = parsed_args.build
+    if not build:
+        if parsed_args.debug:
+            build = "debug"
+        else:
+            build = "release"
+
+    # Get some information from planner_args before it is deleted in split_planner_args().
+    planner_args = parsed_args.planner_args
+    num_planner_args = len(planner_args)
+    mode_switches = ["--translate-options", "--search-options"]
+    last_option_was_mode_switch = planner_args and (planner_args[-1] in mode_switches)
+    double_dash_in_options = "--" in planner_args
+
+    current_mode = util.split_planner_args(parsed_args)
+    num_filenames = len(parsed_args.filenames)
+    has_only_filename_options = (num_filenames == num_planner_args)
+    has_only_filename_or_double_dash_options = (num_filenames + int(double_dash_in_options) == num_planner_args)
+    can_use_double_dash = (1 <= num_planner_args <= 2) and has_only_filename_or_double_dash_options
```

Comment:
It's ... complicated. But I guess we should document this better. 

The double dash can be used to separate filename options from the other options. Say your domain file is called `--debug` for some reason and your problem file is obviously called `--`. To call the planner correctly, you'd use `./fast-downward.py --debug -- -- --search "astar(lmcut)"` The last double dash is removed and everything before it is treated as a filename. This gets complicated because our partial parse might only have the first double dash that is meant to be a filename but since it is currently the last double dash, it is considered a separator. So even though we already have a separator, we still have to suggest one  because there is an interpretation of the command line where we could still need one. Together with the fact that we can take either one or two input files, this creates this mess.

Reply:
In case it's not generally known: "--" acting as a separator between options (which always start with "-") and other arguments (which only rarely start with "--", but are in principle unrestricted) is a general though not universal Unix command-line parsing feature. Tools like argparse handle "--" in this fashion automatically, and because this is expected behavior for all "modern" programs, it's not usually documented to the user.

In the driver we have the special situation that we have to infer *two* separators in the command-line options (between driver options and filenames and between filenames and component options), which is why we extend this convention further.